src/main/java/seedu/ecardnomics/game/GameEngine.java

```diff
@@ -0,0 +1,139 @@
+package seedu.ecardnomics.game;
+
+import seedu.ecardnomics.Ui;
+import seedu.ecardnomics.command.Command;
+import seedu.ecardnomics.command.ExitCommand;
+import seedu.ecardnomics.command.game.DoneGameCommand;
+import seedu.ecardnomics.command.game.GameResponseCommand;
+import seedu.ecardnomics.deck.FlashCard;
+import seedu.ecardnomics.parser.GameParser;
+
+public class GameEngine {
+    GameStorage storage;
+    GameParser gameParser;
+
+    GameEngine(GameStorage storage) {
+        gameParser = new GameParser(storage.originalDeck);
+        this.storage = storage;
+    }
+
+    Command runGameLoop() {
+        Command command;
+
+        do {
+            FlashCard flashCard = poseQuestion();
+            command = getAttempt();
+            if (isTerminate(command)) {
+                break;
+            }
+            Ui.printAnswerGameMode(flashCard.getAnswer());
+            try {
+                Ui.printAttemptFeedback(checkResponse(command, flashCard));
+            } catch (Exception e) {
+                e.getMessage();
+            }
+            command = update(getInclExclYorNResponse(), flashCard, command);
+
+        } while (!isTerminate(command) && !isEndOfDeque());
+
+        return command;
+    }
+
+    Command update(String YorN, FlashCard flashCard, Command command) {
+        if (isTerminate(command)) {
+            return command;
+        }
```

Comment:
Should we remove this branch since the loop will break at line 27 and update would not be called? Can probably replace it with an assert !isTerminate(command).

Reply:
Oh yea definitely. I wrote this before I finalised the runGameLoop. Thanks for the great catch!